Guide new users to add `use super::*;` to `mod test`.

### Problem

`cargo init --lib` produces a src/lib.rs file contaning this:
```rust
#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        let result = 2 + 2;
        assert_eq!(result, 4);
    }
}
```
When I was first learning Rust, I'd start with this code, and add a new function, and a unit test, like this:
```rust
pub fn add_numbers(a: i32, b: i32) -> i32 {
    a + b
}

#[cfg(test)]
mod tests {
    #[test] 
    fn it_works() {
        let result = add_numbers(2, 2);
        assert_eq!(result, 4);
    }
}
```
But, this doesn't compile:
```
error[E0425]: cannot find function `add_numbers` in this scope
 --> src/lib.rs:9:22
  |
9 |         let result = add_numbers(2, 2);
  |                      ^^^^^^^^^^^ not found in this scope
  |
help: consider importing this function
  |
8 |     use crate::add_numbers;
  |
```
So first, this in itself is a mild papercut&mdash;the seemingly obvious thing didn't work. But next, if I would follow rustc's suggestions, or let IDEs fix this for me, I'd eventually end up with a big pile of `use`s, as I write more code and add tests for it. This would add more papercuts: clutter, and a focal point for merge conflicts.

It would almost always be better for me to use `use super::*` inside the `mod tests`, because once that's added, it avoids these papercuts from then on. So the feature request is: can we do more to guide users to `use super::*;`?

The [relevant documentation](https://doc.rust-lang.org/book/ch11-01-writing-tests.html) does recommend using `use super::*;`, but when I was learning Rust, I didn't read that section because I knew what an "assert" was from other languages and didn't think I needed to read that whole section in the docs, so I missed the `use super::*;` advice within.

As a historical note, these papercuts contributed to my forming a habit of writing unit tests without `mod tests`, which is what prompted the discussion which led to #10531.

### Proposed Solution

Adding `use super::*;` to the current code would get an unused-import warning. So one option would be to add it, but commented out, like this:

```rust
#[cfg(test)]
mod tests {
    // Uncomment this to use items from the containing module.
    //use super::*;

    #[test]
    fn it_works() {
        let result = 2 + 2;
        assert_eq!(result, 4);
    }
}
```


### Notes

_No response_

--- a/src/cargo/ops/cargo_new.rs
+++ b/src/cargo/ops/cargo_new.rs
@@ -809,13 +809,17 @@ fn main() {
 "
         } else {
             b"\
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+    
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/src/cargo/ops/cargo_new.rs
+++ b/src/cargo/ops/cargo_new.rs
@@ -811,6 +811,8 @@ fn main() {
             b"\
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/src/cargo/ops/cargo_new.rs
+++ b/src/cargo/ops/cargo_new.rs
@@ -816,7 +816,7 @@ pub fn add(left: usize, right: usize) -> usize {
 #[cfg(test)]
 mod tests {
     use super::*;
-    
+
     #[test]
     fn it_works() {
         let result = add(2, 2);

--- a/tests/testsuite/init/auto_git/out/src/lib.rs
+++ b/tests/testsuite/init/auto_git/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/auto_git/out/src/lib.rs
+++ b/tests/testsuite/init/auto_git/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/formats_source/out/src/lib.rs
+++ b/tests/testsuite/init/formats_source/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+  left + right
+}
+
 #[cfg(test)]
 mod tests {
-  // Uncomment this to use items from the containing module.
-  // use super::*;
+  use super::*;
+
   #[test]
   fn it_works() {
-    let result = 2 + 2;
+    let result = add(2, 2);
     assert_eq!(result, 4);
   }
 }

--- a/tests/testsuite/init/formats_source/out/src/lib.rs
+++ b/tests/testsuite/init/formats_source/out/src/lib.rs
@@ -1,10 +1,10 @@
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
-    #[test]
-    fn it_works() {
-        let result = 2 + 2;
-        assert_eq!(result, 4);
-    }
+  // Uncomment this to use items from the containing module.
+  // use super::*;
+  #[test]
+  fn it_works() {
+    let result = 2 + 2;
+    assert_eq!(result, 4);
+  }
 }

--- a/tests/testsuite/init/formats_source/out/src/lib.rs
+++ b/tests/testsuite/init/formats_source/out/src/lib.rs
@@ -1,8 +1,10 @@
 #[cfg(test)]
 mod tests {
-  #[test]
-  fn it_works() {
-    let result = 2 + 2;
-    assert_eq!(result, 4);
-  }
+    // Uncomment this to use items from the containing module.
+    // use super::*;
+    #[test]
+    fn it_works() {
+        let result = 2 + 2;
+        assert_eq!(result, 4);
+    }
 }

--- a/tests/testsuite/init/fossil_autodetect/out/src/lib.rs
+++ b/tests/testsuite/init/fossil_autodetect/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/fossil_autodetect/out/src/lib.rs
+++ b/tests/testsuite/init/fossil_autodetect/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/git_autodetect/out/src/lib.rs
+++ b/tests/testsuite/init/git_autodetect/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/git_autodetect/out/src/lib.rs
+++ b/tests/testsuite/init/git_autodetect/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/git_ignore_exists_no_conflicting_entries/out/src/lib.rs
+++ b/tests/testsuite/init/git_ignore_exists_no_conflicting_entries/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/git_ignore_exists_no_conflicting_entries/out/src/lib.rs
+++ b/tests/testsuite/init/git_ignore_exists_no_conflicting_entries/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/ignores_failure_to_format_source/out/src/lib.rs
+++ b/tests/testsuite/init/ignores_failure_to_format_source/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/ignores_failure_to_format_source/out/src/lib.rs
+++ b/tests/testsuite/init/ignores_failure_to_format_source/out/src/lib.rs
@@ -5,7 +5,7 @@ pub fn add(left: usize, right: usize) -> usize {
 #[cfg(test)]
 mod tests {
     use super::*;
-
+    
     #[test]
     fn it_works() {
         let result = add(2, 2);

--- a/tests/testsuite/init/ignores_failure_to_format_source/out/src/lib.rs
+++ b/tests/testsuite/init/ignores_failure_to_format_source/out/src/lib.rs
@@ -5,7 +5,7 @@ pub fn add(left: usize, right: usize) -> usize {
 #[cfg(test)]
 mod tests {
     use super::*;
-    
+
     #[test]
     fn it_works() {
         let result = add(2, 2);

--- a/tests/testsuite/init/ignores_failure_to_format_source/out/src/lib.rs
+++ b/tests/testsuite/init/ignores_failure_to_format_source/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/mercurial_autodetect/out/src/lib.rs
+++ b/tests/testsuite/init/mercurial_autodetect/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/mercurial_autodetect/out/src/lib.rs
+++ b/tests/testsuite/init/mercurial_autodetect/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/pijul_autodetect/out/src/lib.rs
+++ b/tests/testsuite/init/pijul_autodetect/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/pijul_autodetect/out/src/lib.rs
+++ b/tests/testsuite/init/pijul_autodetect/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/simple_git/out/src/lib.rs
+++ b/tests/testsuite/init/simple_git/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/simple_git/out/src/lib.rs
+++ b/tests/testsuite/init/simple_git/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/simple_git_ignore_exists/out/src/lib.rs
+++ b/tests/testsuite/init/simple_git_ignore_exists/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/simple_git_ignore_exists/out/src/lib.rs
+++ b/tests/testsuite/init/simple_git_ignore_exists/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/simple_hg/out/src/lib.rs
+++ b/tests/testsuite/init/simple_hg/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/simple_hg/out/src/lib.rs
+++ b/tests/testsuite/init/simple_hg/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/simple_hg_ignore_exists/out/src/lib.rs
+++ b/tests/testsuite/init/simple_hg_ignore_exists/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/simple_hg_ignore_exists/out/src/lib.rs
+++ b/tests/testsuite/init/simple_hg_ignore_exists/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/init/simple_lib/out/src/lib.rs
+++ b/tests/testsuite/init/simple_lib/out/src/lib.rs
@@ -1,5 +1,7 @@
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/init/simple_lib/out/src/lib.rs
+++ b/tests/testsuite/init/simple_lib/out/src/lib.rs
@@ -1,10 +1,14 @@
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

--- a/tests/testsuite/new.rs
+++ b/tests/testsuite/new.rs
@@ -41,6 +41,8 @@ fn simple_lib() {
         contents,
         r#"#[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
     #[test]
     fn it_works() {
         let result = 2 + 2;

--- a/tests/testsuite/new.rs
+++ b/tests/testsuite/new.rs
@@ -39,13 +39,17 @@ fn simple_lib() {
     let contents = fs::read_to_string(&lib).unwrap();
     assert_eq!(
         contents,
-        r#"#[cfg(test)]
+        r#"pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
+#[cfg(test)]
 mod tests {
-    // Uncomment this to use items from the containing module.
-    // use super::*;
+    use super::*;
+
     #[test]
     fn it_works() {
-        let result = 2 + 2;
+        let result = add(2, 2);
         assert_eq!(result, 4);
     }
 }

